realm/src/main/java/io/realm/internal/Group.java

```diff
@@ -287,17 +253,27 @@ public String toJson() {
         return nativeToJson(nativePtr);
     }
 
-    protected native String nativeToJson(long nativeGroupPtr);
-
     public String toString() {
         return nativeToString(nativePtr);
     }
 
-    protected native void nativeCommit(long nativeGroupPtr);
-
-    protected native String nativeToString(long nativeGroupPtr);
-
     private void throwImmutable() {
         throw new IllegalStateException("Objects cannot be changed outside a transaction; see beginTransaction() for details.");
     }
+
+    protected native long createNative();
+    protected native long createNative(String filepath, int value);
+    protected native long createNative(byte[] data);
+    protected native long createNative(ByteBuffer buffer);
+    protected static native void nativeClose(long nativeGroupPtr);
```

Comment:
For some reason close is called from Context in static way, so unfortunately there is no way around static here. Otherwise I would agree with @nhachicha.